lib/java/com/google/android/material/card/MaterialCardView.java

```diff
@@ -93,6 +96,49 @@
   private static final String LOG_TAG = "MaterialCardView";
   private static final String ACCESSIBILITY_CLASS_NAME = "androidx.cardview.widget.CardView";
 
+  /**
+   * Gravity used to position the checked icon at the top|start of the Card.
+   *
+   * @see #setCheckedIconGravity(int)
+   * @see #getCheckedIconGravity()
+   */
+  public static final int CHECKED_ICON_GRAVITY_TOP_START = 0x1;
+
+  /**
+   * Gravity used to position the checked icon at the bottom|start of the Card.
+   *
+   * @see #setCheckedIconGravity(int)
+   * @see #getCheckedIconGravity()
+   */
+  public static final int CHECKED_ICON_GRAVITY_BOTTOM_START = 0x2;
+
+  /**
+   * Gravity used to position the checked icon at the top|end of the Card.
+   *
+   * @see #setCheckedIconGravity(int)
+   * @see #getCheckedIconGravity()
+   */
+  public static final int CHECKED_ICON_GRAVITY_TOP_END = 0x3;
+
+  /**
+   * Gravity used to position the checked icon at the bottom|end of the Card.
+   *
+   * @see #setCheckedIconGravity(int)
+   * @see #getCheckedIconGravity()
+   */
+  public static final int CHECKED_ICON_GRAVITY_BOTTOM_END = 0x4;
+
+  /** Positions the icon can be set to. */
+  @IntDef({
+      CHECKED_ICON_GRAVITY_TOP_START,
+      CHECKED_ICON_GRAVITY_BOTTOM_START,
+      CHECKED_ICON_GRAVITY_TOP_END,
+      CHECKED_ICON_GRAVITY_BOTTOM_END
+  })
```

Comment:
I think it is better to have both the vertical and horizontal gravity not the mixture of both and also center alignment would be helpful